### ISSUE TITLE
Image flickering when removed cache

### ISSUE THREAD

**Issue by akaakoz** [SDWebImage/SDWebImage#3514]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos
 Platform Version        | e.g. 16.3.1 / 12.6.3
 SDWebImage Version      | e.g. 5.15.5
 Integration Method      | e.g. cocoapods
 Xcode Version           | e.g. Xcode 14.2
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Im setting the image like below
```
let options = SDWEBImageOptions = [.refreshcached. retryFailed]
sd_image(with: url, placeHolderImage: UIImage, options: options)
```

And every time I reloadData() inside tableView by pulling to refresh(The image already has been set inside `viewDidLoad()`), the image gets flickering when I clear the cache using `SDWEBImageCache.shared.removeImage(forkey: String, fromDisk: true)`. 

How can I prevent that? or Shouldn't I clear the cache every time the image is set?
